src/sql_engine/src/query/scalar.rs

```diff
@@ -12,26 +12,55 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-///! Module for representing scalar level operations. Implementation of
-///! theses operators will be defined in a sperate module.
-use super::{ColumnType, RelationType, Row};
+use sqlparser::ast::BinaryOperator;
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
-pub enum BinaryOp {}
+use representation::{Datum, ScalarType};
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
-pub enum UnaryOp {}
+///! Module for representing scalar level operations. Implementation of
+///! theses operators will be defined in a sperate module.
+// use crate::query::relation::RelationType;
 
 /// Operation performed on the table
 /// influenced by Materialized's ScalarExpr
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub enum ScalarOp {
     /// column access
-    Column(usize),
+    Column(usize, ScalarType),
     /// literal value (owned) and expected type.
-    Literal(Vec<Row>, RelationType),
+    Literal(Datum<'static>),
     /// binary operator
-    Binary(BinaryOp, Box<ScalarOp>, Box<ScalarOp>),
-    /// uanry operator
-    Unary(UnaryOp, Box<ScalarOp>),
+    Binary(BinaryOperator, Box<ScalarOp>, Box<ScalarOp>, ScalarType),
+    /// unary operator
+    // Unary(UnaryOperator, Box<ScalarOp>, ScalarType),
```

Comment:
Could you remove commented code, and docs too? Otherwise, `Assignment` is a `unary operator` according to doc section 😄